tests/integration/bulk_import/test_bulk_import_request.py

```diff
@@ -0,0 +1,139 @@
+
+    assert bulk_import_request.project() == configured_project
+    assert bulk_import_request.name == name
+    assert bulk_import_request.error_file_url is None
+    assert bulk_import_request.status_file_url is None
+    assert bulk_import_request.state == BulkImportRequestState.RUNNING
+    __assert_file_content(bulk_import_request.input_file_url, predictions)
+
+
+def test_get(client, configured_project):
+    name = str(uuid.uuid4())
+    url = "https://storage.googleapis.com/labelbox-public-bucket/predictions_test_v2.ndjson"
+    configured_project.upload_annotations(name=name,
+                                          annotations=url,
+                                          validate=False)
+
+    bulk_import_request = BulkImportRequest.from_name(
+        client, project_id=configured_project.uid, name=name)
+
+    assert bulk_import_request.project() == configured_project
+    assert bulk_import_request.name == name
+    assert bulk_import_request.input_file_url == url
+    assert bulk_import_request.error_file_url is None
+    assert bulk_import_request.status_file_url is None
+    assert bulk_import_request.state == BulkImportRequestState.RUNNING
+
+
+def test_validate_ndjson(tmp_path, configured_project):
+    file_name = f"broken.ndjson"
+    file_path = tmp_path / file_name
+    with file_path.open("w") as f:
+        f.write("test")
+
+    with pytest.raises(ValueError):
+        configured_project.upload_annotations(name="name",
+                                              annotations=str(file_path))
+
+
+def test_validate_ndjson_uuid(tmp_path, configured_project, predictions):
+    file_name = f"repeat_uuid.ndjson"
+    file_path = tmp_path / file_name
+    repeat_uuid = predictions.copy()
+    uid = str(uuid.uuid4())
+    repeat_uuid[0]['uuid'] = uid
+    repeat_uuid[1]['uuid'] = uid
+
+    with file_path.open("w") as f:
+        ndjson.dump(repeat_uuid, f)
+
+    with pytest.raises(UuidError):
+        configured_project.upload_annotations(name="name",
+                                              annotations=str(file_path))
+
+    with pytest.raises(UuidError):
+        configured_project.upload_annotations(name="name",
+                                              annotations=repeat_uuid)
+
+
+@pytest.mark.slow
+def test_wait_till_done(configured_project):
+    name = str(uuid.uuid4())
+    url = "https://storage.googleapis.com/labelbox-public-bucket/predictions_test_v2.ndjson"
+    bulk_import_request = configured_project.upload_annotations(name=name,
+                                                                annotations=url,
+                                                                validate=False)
+
+    bulk_import_request.wait_until_done()
+
+    assert (bulk_import_request.state == BulkImportRequestState.FINISHED or
+            bulk_import_request.state == BulkImportRequestState.FAILED)
+
+
+def __assert_file_content(url: str, predictions):
```

Comment:
Why the double underscore?

Reply:
I didn't add this code. I just moved it to the new directory.

Reply:
I'm happy to change it if you want me to.

Reply:
Please do! I always try to do these kind of minor improvements in the code I touch, in time it accumulates.